akka-cluster/src/main/scala/akka/cluster/Gossip.scala

```diff
@@ -154,10 +154,9 @@ private[cluster] final case class Gossip(
 
     // 1. merge sets of tombstones
     val mergedTombstones = tombstones ++ that.tombstones
-    val newTombstonedNodes = mergedTombstones.keySet diff that.tombstones.keySet
```

Comment:
Aaah the old approach would also have worked if this were `mergedTombstones.keySet diff tombstones.keySet` I suppose?

Anyway this is more safe.